SleepEngine/Engine/Jobs/JobSystem.cpp

```diff
@@ -0,0 +1,86 @@
+#include "stdafx.h"
+#include "JobSystem.h"
+
+BEGIN_NAMESPACE_SLEEP
+
+namespace
+{
+    constexpr unsigned ThreadsLeftAlone = 1;
+}
+
+JobSystem::JobSystem()
+    : JobSystem(JobSystem::getWorkerThreadCount())
+{
+}
+
+JobSystem::JobSystem(size_t workerThreadCount)
+    : m_jobAvailableEvents(workerThreadCount)
+{
+    if (workerThreadCount == 0)
+    {
+        return;
+    }
+
+    // TODO: add render thread
+    JobQueue& genericQueue = m_affinityToQueues[JobAffinity::Generic];
+
+    m_jobThreads.reserve(workerThreadCount);
+    for (size_t i = 0; i < workerThreadCount; i++)
+    {
+        m_jobThreads.emplace_back(genericQueue, m_jobAvailableEvents.at(i), m_shutdownRequested);

```

Comment:
are you sure m_jobAvailableEvents.at won't throw exception here?

Reply:
yes, it's initialized with the same count in the init list